### PR TITLE
pthreadpool: modernize more (and generate v2 packages)

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
```
ERROR: Missing prebuilt package for 'fxdiv/cci.20200417'
```

@uilianries @danimtb missing prebuild package here also

**Review by prince-chrismc — DISMISSED** [review]

File: recipes/pthreadpool/all/conanfile.py [hunk 1]
```diff
@@ -36,35 +36,24 @@ def config_options(self):
...
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
 
     def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("fxdiv/cci.20200417")
+        self.requires("fxdiv/cci.20200417", transitive_headers=True)
```
Comment (jcar87) [inline recipes/pthreadpool/all/conanfile.py:47]: Is this the case?
The single public header from `pthreadpool`, does not have a transitive header dependency on `fxdiv`, I can only see references to:
```
#include <stddef.h>
#include <stdint.h>
```

On the other hand, the library authors explicitly mark it as a _private_ dependency:
https://github.com/Maratyszcza/pthreadpool/blob/master/CMakeLists.txt#L164

So I'm unsure as to what motivates this change.

**Review by jcar87 — CHANGES_REQUESTED** [review]

File: recipes/pthreadpool/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/pthreadpool/all/conanfile.py:47]: Yes, you're right: https://github.com/Maratyszcza/pthreadpool/blob/b4589998be9a0f794236cf46f1b5b232b2b15ca3/include/pthreadpool.h (this is the commit packaged here)

**Review by jcar87 — APPROVED** [review]
Thanks for your contribution!

**Review by davidsanfal — APPROVED** [review]

**Pull request merged**